search/fibonacci_search.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright 2020 @author sprintyaf
+ * @file
+ * @brief [Fibonacci search
+ * algorithm](https://en.wikipedia.org/wiki/Fibonacci_search_technique)
+ */
+
+#include <iostream>
+#include <vector> // for std::vector class
+#include <cassert> // for assert
+
+
+
+/**
+ * Returns minimum of the two numbers
+ */
+int min(int first, int second){
+    if(first < second){
+        return first;
+    } else {
+        return second;
```

Comment:
You can shorten this block by using the ternary operator like so:

```c++
int min(int first, int second) {
    return first < second ? first : second;
}
```
Of course this is a matter of style and readability.

Reply:
or simply use the standard C++ library:
`std::min(first, second);`